### ISSUE TITLE
Add `jax.numpy`  mathematical functions to Jax frontend

### ISSUE THREAD

**Issue by hirwa-nshuti** [unifyai/ivy#8426]
These functions can be found in [`jax.numpy`](https://jax.readthedocs.io/en/latest/jax.numpy.html) namespace

- [x] #4884
- [x] #4915
- [x] #4916
- [ ] angle
- [x] #5248
- [x] #5249
- [x] #5250
- [x] #5251
- [x] #5369
- [x] #5371
- [x] #6895
- [ ] around
- [ ] cbrt
- [x] #6326
- [x] #4957
- [ ] conj
- [ ] conjugate
- [ ] convolve
- [ ] copysign
- [x] cos
- [ ] cosh
- [x] #6467
- [ ] #7477
- [ ] diff
- [x] #7771
- [ ] divmod
- [x] #4946
- [ ] ediff1d
- [ ] einsum_path
- [ ] exp
- [x] #6471
- [ ] expm1
- [ ] fabs
- [ ] fix
- [x] #6321
- [x] #5691
- [ ] floor_divide
- [x] #5705
- [ ] fmin
- [ ] fmod
- [ ] frexp
- [x] #6532
- [ ] gradient
- [x] #6458
- [ ] hypot
- [x] #6561
- [ ] imag
- [ ] inner
- [ ] interp
- [x] #6584
- [x] #6599
- [ ] ldexp
- [ ] log
- [x] #7277
- [x] #5895
- [ ] #5896
- [x] #7286
- [x] #6602
- [ ] matmul
- [x] #7049
- [x] #6901 
- [x] #4953
- [ ] #6900
- [ ] mod
- [ ] modf
- [x] #6899
- [ ] nan_to_num
- [ ] nanprod
- [x] #7711
- [ ] nextafter
- [ ] outer
- [ ] poly
- [ ] polyadd
- [ ] polyder
- [ ] polydiv
- [ ] polyfit
- [ ] polyint
- [ ] polymul
- [ ] polysub
- [ ] polyval
- [ ] positive
- [x] #6082
- [ ] prod
- [ ] product
- [x] #7749
- [ ] radians
- [ ] real
- [ ] reciprocal
- [ ] remainder
- [ ] rint
- [ ] roots
- [ ] round
- [ ] round_
- [ ] sign
- [ ] signbit
- [x] #6896
- [ ] sinc
- [x] #6897
- [x] #7854
- [ ] square
- [ ] #5300
- [x] #6317
- [x] #7165
- [x] #7166
- [x] #7755
- [ ] trace
- [x] #7855
- [x] #6618
- [x] #7776
- [x] #6330
- [ ] unwrap
- [ ] vdot

**Comment by adnit** [comment]
- [ ] log#9060

**Comment by adnit** [comment]
9060

**Comment by Jacob6819** [comment]
#9092

**Comment by Jacob6819** [comment]
#9093

**Comment by aymuos15** [comment]
- [ ] #9477

**Comment by hmahmood24** [comment]
- [ ] #9508

**Comment by hmahmood24** [comment]
- [ ] #9510

**Comment by jiisa-k** [comment]
#9525

**Comment by Aidan7757** [comment]
- [angle] #9602

**Comment by moloydas** [comment]
around #9697

**Comment by glasalm** [comment]
- [ ] https://github.com/unifyai/ivy/issues/9704

**Comment by moloydas** [comment]
#9735

**Comment by moloydas** [comment]
#9736

**Comment by TakshPanchal** [comment]
- [ ] #9743

**Comment by TakshPanchal** [comment]
#9744

**Comment by YanniPapandreou** [comment]
- [ ] #9756

**Comment by hirwa-nshuti** [comment]
> The following tests are not implemented:
> 
> * [x]  arcsin
> * [x]  log1p

Hey would you add them😊

**Comment by karalleyna** [comment]
> > The following tests are not implemented:
> > 
> > * [x]   arcsin
> > * [x]   log1p
> 
> Hey would you add them😊

I already implemented them :)

**Comment by andylolu2** [comment]
- [ ] #9873

**Comment by AhmedHashish123** [comment]
- [ ] #9922

**Comment by manuelsolan-o** [comment]
round

**Comment by leacornelio** [comment]
- [ ] #10309

**Comment by vaithak** [comment]
- [ ] #10402

**Comment by JPGoodale** [comment]
- [#10499]

**Comment by jatinS-dev** [comment]
- [ ] #10612

**Comment by kunalagra** [comment]
- [ ] #10675

**Comment by thegeek13242** [comment]
nanprod #10922

**Comment by vandit98** [comment]
roll #10889

**Comment by lucasalavapena** [comment]
- [ ] #10981

**Comment by lucasalavapena** [comment]
Okay, apparently `real` is already implemented (See https://github.com/unifyai/ivy/commit/1f961d73158c261b2b04372c1121e1a51b1b15ad) even though there are no indication in this issue. Here is the  relevant [issue](https://github.com/unifyai/ivy/issues/9736), based on GitHub it was referenced here so perhaps an entry issue or less likely there is an issue with the bot?

**Comment by lucasalavapena** [comment]
- [ ] #10992

**Comment by usama-baloch** [comment]
#11004

**Comment by usama-baloch** [comment]
#11012

**Comment by BrookeMa** [comment]
- [ ] #11072

**Comment by RoryR99** [comment]
- [ reciprocal] #8426

**Comment by RoryR99** [comment]
[reciprocal] #11868

**Comment by RoryR99** [comment]
- [reciprocal] #11868

**Comment by raeesm1200** [comment]
- [ ] #12357

**Comment by RoryR99** [comment]
- [ ] Radians 
- [ ] Conjugate

**Comment by x0axz** [comment]
- [ ] #12367

**Comment by x0axz** [comment]
- [ ] #12370

**Comment by Moamen-Elsayed** [comment]
#12385

**Comment by RoryR99** [comment]
- [ ] https://github.com/unifyai/ivy/issues/12364

**Comment by xingyu-long** [comment]
- [ ] #12528

**Comment by eigengravy** [comment]
#12734

**Comment by Ajitofy** [comment]
- [ ] imag #13011

**Comment by Ajitofy** [comment]
- [ ] #13062

**Comment by Ajitofy** [comment]
- [ ] #13063

**Comment by gshreya5** [comment]
gradient #13096

**Comment by sharjeelnawaz8182** [comment]
- [ ] #13011

**Comment by sondertg** [comment]
- [ ] #13688

**Comment by KarimIbrahim11** [comment]
- [ ] #13697

**Comment by KarimIbrahim11** [comment]
- [ ] #13700

**Comment by abhijeetsharma200** [comment]
- [ ] #13870

**Comment by vandit98** [comment]
real #13871

**Comment by sondertg** [comment]
- [ ] #13883

**Comment by sondertg** [comment]
- [ ] #13885

**Comment by sondertg** [comment]
- [ ] #13886

**Comment by KevinUli** [comment]
- [ ] #14485

**Comment by RakshitKumar04** [comment]
- [ ] #14522

**Comment by hammadjka** [comment]
- [ ] #14556

**Comment by hammadjka** [comment]
> * [ ]  [conjugate #14556](https://github.com/unifyai/ivy/issues/14556)

Please correct me If I'm wrong but I just realized that conjugate is not a part of jax.numpy but jax.lax. As such I don't believe this is the right ToDo list for this sub task.

**Comment by ZnMind** [comment]
#14822

**Comment by natek-1** [comment]
- [ ] #15721

**Comment by natek-1** [comment]
- [ ] #15765

**Comment by ahmadel1** [comment]
#16106

**Comment by ahmadel1** [comment]
#16106

**Comment by lfcalmeida** [comment]
#16279

**Comment by natek-1** [comment]
- [ ] #16465

**Comment by OziomaEunice** [comment]
- [ ] #16559

**Comment by Gwyn-The-Human** [comment]
#17019

**Comment by umilic16** [comment]
#18237

**Comment by jdc98** [comment]
- [ ] #18356

**Comment by moezAhmad** [comment]
[imag]#18981

**Comment by ionorins** [comment]
- [ ] #19467

**Comment by OkeyAmy** [comment]
[cos] #19833

**Comment by AryanSharma21** [comment]
#20892

**Comment by a0m0rajab** [comment]
@AryanSharma21 @OkeyAmy  in order to get your comments added to the list you need to follow this guides: https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues

What you are missing is this: 

> Comment on the ToDo list issue with a reference to this issue like so:
> `- [ ] #Issue_number`

**Comment by AryanSharma21** [comment]
Hey @a0m0rajab  thanks for the help

**Comment by a0m0rajab** [comment]
@AryanSharma21 you are welcome, I think you need to have an empty space as well after brackets.

**Comment by AryanSharma21** [comment]
- [ ] #20892

**Comment by spidy1996** [comment]
#20942

**Comment by a0m0rajab** [comment]
@spidy1996 which tasks are completed that you noticed?

**Comment by OkeyAmy** [comment]
"Cos" function

**Comment by OkeyAmy** [comment]
"cos" function

**Comment by spidy1996** [comment]
> @spidy1996 which tasks are completed that you noticed?

Cos, cosh, exp, mod, round. There would be others as well, I'm sure of it.

**Comment by a0m0rajab** [comment]
> > @spidy1996 which tasks are completed that you noticed?
> 
> > @a0m0rajab Cos, cosh, exp, mod, round. There would be others as well, I'm sure of it.

I think it's okay since it has been checked in the checklist: 
![image](https://github.com/unifyai/ivy/assets/18273833/db10e9b3-5610-4dd0-a306-7070f4354259)

There is no need to connect with it's related issue (doing this would be better though but yeah)

**Comment by OzMoti** [comment]
- [ ] #20967

**Comment by Tonycrux** [comment]
- [ ] #21017

**Comment by Tonycrux** [comment]
- [ ] #21033

**Comment by Mwangi-ke** [comment]
#21787

**Comment by axcel342** [comment]
-[]#21017

**Comment by axcel342** [comment]
- [ ] #21017

**Comment by axcel342** [comment]
#22611

**Comment by AbdullahSabry** [comment]
- [ ] #22697

**Comment by AbdullahSabry** [comment]
- [ ] #22699

**Comment by dash96** [comment]
- [ ] roots #23179

**Comment by rohitkg83** [comment]
[] - #23797

**Comment by rohitkg83** [comment]
-[] #25767

**Comment by thelonewolff** [comment]
https://github.com/unifyai/ivy/issues/25811

**Comment by OkeyAmy** [comment]
- [ ] #26054

**Comment by ChibuezeOnejeme** [comment]
#26419

**Comment by ChibuezeOnejeme** [comment]
- [ ]https://github.com/unifyai/ivy/issues/26419

**Comment by ro4i7** [comment]
#26421

**Comment by ro4i7** [comment]
#26422

**Comment by ro4i7** [comment]
#26421

**Comment by rohitkg83** [comment]
-[] #26432

**Comment by Prathamsahu52** [comment]
- [ ] #26452

**Comment by 20481A05K0** [comment]
hey there
may i know the problem clearly actually maximum all mathematical functions are implemented then what is the issue

**Comment by JeffreyJu8** [comment]
#26528

**Comment by 20481A05K0** [comment]
Hello there!!
does gradient is implemented in jax frontend or can i start working on it please respond asap
